fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoteScanPartitionPruneRule.java

```diff
@@ -62,9 +65,16 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
         Map<ColumnRefOperator, Set<Long>> columnToNullPartitions = Maps.newHashMap();
 
         try {
-            initPartitionInfo(operator, columnToPartitionValuesMap, columnToNullPartitions, context);
-            classifyConjuncts(operator, columnToPartitionValuesMap);
-            computePartitionInfo(operator, columnToPartitionValuesMap, columnToNullPartitions, context);
+            if (operator.getTable() instanceof IcebergTable) {
+                // A little trick to set min max for iceberg table here
+                classifyConjuncts(operator, columnToPartitionValuesMap);
```

Comment:
Could this piece of code be integrated into the following framework? So there is no if/else stmt.

```
initPartitionInfo(operator, columnToPartitionValuesMap, columnToNullPartitions, context);
                classifyConjuncts(operator, columnToPartitionValuesMap);
                computePartitionInfo(operator, columnToPartitionValuesMap, columnToNullPartitions, context);
```